src/components/InteractionLayer/index.js

```diff
@@ -236,12 +285,41 @@ class InteractionLayer extends React.Component {
       );
     }
     const annotations = this.props.annotations.map(a => (
-      <Annotation key={a.id} {...a} xScale={xScale} height={height} />
+      <Annotation key={a.id} {...a} height={height} />
     ));
+    const xScale = createXScale(this.props.subDomain, width);
+    const areas = this.props.areas.map(a => {
+      let a1 = a;
+      let s = null;
+      if (a.seriesId) {
+        s = series.find(s1 => s1.id === a.seriesId);
+      }
+      if (a.start.xval && a.end.xval) {
+        const yScale = createYScale(s.yDomain, height);
+        a1 = {
+          ...a,
```

Comment:
only seriesId is needed instead of a spread?

Reply:
If the users passes in other data, I don't see why we shouldn't keep it. But I'm indifferent to this -- you or @martincognite feel strongly one way or the other?

Reply:
I can't actually remember why I thought there was an issue here, but right now it looks good to me as is.

Reply:
That's what I like to hear.